vividus-plugin-web-app/src/main/java/org/vividus/selenium/screenshot/ScreenshotTaker.java

```diff
@@ -154,14 +154,10 @@ public void takeScreenshot(Path screenshotFilePath, boolean viewportScreenshot)
 
     private Optional<Screenshot> createScreenshot(byte[] screenshotData, String screenshotName)
     {
-        if (screenshotData.length > 0)
-        {
-            Screenshot screenshot = new Screenshot();
-            screenshot.setData(screenshotData);
-            screenshot.setFileName(generateScreenshotFileName(screenshotName));
-            return Optional.of(screenshot);
-        }
-        return Optional.empty();
+        Screenshot screenshot = screenshotData.length > 0
+                ? new Screenshot(generateScreenshotFileName(screenshotName), screenshotData)
+                : null;
+        return Optional.ofNullable(screenshot);
```

Comment:
```java
return screenshotData.length > 0
            ? Optional.of(new Screenshot(generateScreenshotFileName(screenshotName), screenshotData))
            : Optional.empty();
```

Reply:
fixed, but imho too many method invocations per line

Reply:
in this case you can keep old `if`